sn_networking/src/replication_fetcher.rs

```diff
@@ -31,51 +31,67 @@ pub(crate) enum HolderStatus {
 #[derive(Default)]
 pub(crate) struct ReplicationFetcher {
     to_be_fetched: BTreeMap<NetworkAddress, BTreeMap<PeerId, (Instant, HolderStatus)>>,
-    on_going_fetches: usize,
+    on_going_fetches: BTreeMap<PeerId, usize>,
```

Comment:
this per `Peer` measure may have a risk of contacting duplicate peers for the same entry,
or too peers contacted concurrently causing self choked when receiving responses.